catalog/async/catalog-async-processingframework/src/main/java/org/codice/ddf/catalog/async/processingframework/impl/InMemoryProcessingFramework.java

```diff
@@ -0,0 +1,294 @@
+                    }
+                }
+            });
+        }
+    }
+
+    private <T extends ProcessResourceItem> void closeInputStream(ProcessRequest<T> request) {
+        request.getProcessItems()
+                .stream()
+                .map(ProcessResourceItem::getProcessResource)
+                .filter(Objects::nonNull)
+                .forEach(resource -> {
+                    try {
+                        resource.getInputStream()
+                                .close();
+                    } catch (IOException e) {
+                        LOGGER.debug(
+                                "Failed to close stream of asynchronous request. There may be temporary files left in data/tmp.");
+                    }
+                });
+    }
+
+    private <T extends ProcessResourceItem> void storeProcessRequest(
+            ProcessRequest<T> processRequest) {
+        LOGGER.trace("Storing update request post processing change(s)");
+
+        Set<ContentItem> contentItemsToUpdate = new HashSet<>();
+        Set<Metacard> metacardsToUpdate = new HashSet<>();
+
+        for (T item : processRequest.getProcessItems()) {
+            if (item.getProcessResource() == null && item.isMetacardModified()) {
+                metacardsToUpdate.add(item.getMetacard());
+            }
+
+            final ProcessResource processResource = item.getProcessResource();
+            if (processResource != null && processResource.isModified()) {
+                try {
+                    byte[] byteArray = IOUtils.toByteArray(processResource.getInputStream());
+                    ByteSource byteSource = ByteSource.wrap(byteArray);
+
+                    ContentItem contentItem = new ContentItemImpl(item.getMetacard()
+                            .getId(),
+                            processResource.getQualifier(),
+                            byteSource,
+                            processResource.getMimeType(),
+                            processResource.getName(),
+                            processResource.getSize(),
+                            item.getMetacard());
+
+                    contentItemsToUpdate.add(contentItem);
+                } catch (IOException e) {
+                    LOGGER.debug("Unable to store process request", e);
+                }
+            }
+        }
+
+        storeContentItemUpdates(contentItemsToUpdate, processRequest.getProperties());
+        storeMetacardUpdates(metacardsToUpdate, processRequest.getProperties());
+    }
+
+    private void storeContentItemUpdates(Set<ContentItem> contentItemsToUpdate,
+            Map<String, Serializable> properties) {
+        if (CollectionUtils.isNotEmpty(contentItemsToUpdate)) {
+            LOGGER.trace("Storing content item updates(s)");
+
+            UpdateStorageRequest updateStorageRequest =
+                    new UpdateStorageRequestImpl(new ArrayList<>(contentItemsToUpdate), properties);
+
+            Security.runAsAdmin(() -> {
```

Comment:
We shouldn't need to elevate here either. Why can't we just pass the subject down that originally created this request in the first place and attach it to the processing request? Then you can just execute this as that user.